### PR TITLE
Chore/improved error surfacing in sql editor

### PR DESCRIPTION
**Pull request by joshenlim**
- SQL editor will now show which line the error is at
- Error line will be scrolled to and highlighted

### CODE REVIEW

File: studio/components/interfaces/SQLEditor/SQLEditor.tsx [hunk 1]
```diff
@@ -171,8 +173,33 @@ const SQLEditor = () => {
...
+      if (id) {
+        if (error.position && monacoRef.current) {
+          const editor = editorRef.current
+          const monaco = monacoRef.current
+
+          const formattedError = error.formattedError ?? ''
+          const lineError = formattedError.slice(formattedError.indexOf('LINE'))
+          const line = Number(lineError.slice(0, lineError.indexOf(':')).split(' ')[1])
+
+          const decorations = editor?.deltaDecorations(
```
Comment (alaister) [inline studio/components/interfaces/SQLEditor/SQLEditor.tsx:186]: Is it worth checking `line` is not `NaN` before calling `deltaDecorations()`?

File: studio/components/interfaces/SQLEditor/SQLEditor.tsx [hunk 2]
```diff
@@ -171,8 +173,33 @@ const SQLEditor = () => {
...
+            [],
+            [
+              {
+                range: new monaco.Range(line, 1, line, 20),
+                options: { isWholeLine: true, inlineClassName: 'bg-amber-800' },
```
Comment (alaister) [inline studio/components/interfaces/SQLEditor/SQLEditor.tsx:191]: `bg-amber-800` vs using red?

File: studio/components/interfaces/SQLEditor/UtilityPanel/UtilityTabResults.tsx [hunk 3]
```diff
@@ -64,12 +64,24 @@ const UtilityTabResults = ({ id, isExecuting }: UtilityTabResultsProps) => {
...
+        <div className="flex flex-row justify-between items-start py-4 px-6">
+          <div>
+            {formattedError.length > 0 ? (
+              formattedError.map((x: string, i: number) => (
+                <pre key={`error-${i}`} className="font-mono text-sm">
+                  {x}
+                </pre>
+              ))
+            ) : (
+              <p className="font-mono text-sm">result.error.error</p>
```
Comment (alaister) [inline studio/components/interfaces/SQLEditor/UtilityPanel/UtilityTabResults.tsx:82]:
```suggestion
              <p className="font-mono text-sm">{result.error.error}</p>
```

**Review by alaister — CHANGES_REQUESTED** [review]
Looking and working great! 🔥

Just a few small things:

File: studio/components/interfaces/SQLEditor/SQLEditor.tsx (hunk 2 above)
Comment (joshenlim) [reply studio/components/interfaces/SQLEditor/SQLEditor.tsx:191]: i think amber is okay - its a warning, not something destructive
lets reserve using red for things that are destructive

File: studio/components/interfaces/SQLEditor/SQLEditor.tsx (hunk 1 above)
Comment (joshenlim) [reply studio/components/interfaces/SQLEditor/SQLEditor.tsx:186]: oh yeah good point!

File: studio/components/interfaces/SQLEditor/SQLEditor.tsx [hunk 4]
```diff
@@ -171,8 +173,35 @@ const SQLEditor = () => {
...
+      if (id) {
+        if (error.position && monacoRef.current) {
+          const editor = editorRef.current
+          const monaco = monacoRef.current
+
+          const formattedError = error.formattedError ?? ''
+          const lineError = formattedError.slice(formattedError.indexOf('LINE'))
+          const line = Number(lineError.slice(0, lineError.indexOf(':')).split(' ')[1])
+
+          if (line) {
```
Comment (alaister) [inline studio/components/interfaces/SQLEditor/SQLEditor.tsx:186]:
```suggestion
          if (!isNaN(line)) {
```

**Review by alaister — APPROVED** [review]
LGTM!

**Pull request merged**